metadata/utils.go

```diff
@@ -204,9 +204,35 @@ func toStringArrayHookFunc() mapstructure.DecodeHookFunc {
 	}
 }
 
+type ComponentTypeValue string
+
+var ComponentType = struct {
```

Comment:
Does this need to be a struct? Can it be a set of constants instead?

As per our conversation on a recent PR, using a struct increases the memory usage, while consts should be easier on the memory.

Reply:
Changed it to constants with an addition `switch` statement check.